charts/consul/templates/ingress-gateways-deployment.yaml

```diff
@@ -26,15 +26,15 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ template "consul.fullname" $root }}-{{ .name }}
+  name: {{ template "consul.fullname" $root }}-{{ .name }}-ingress-gateway
```

Comment:
throughout this PR (and the terminating gateway one), we add `-terminating-gateway` to the end of all resources because unlike other components, users can launch multiple terminating gateways and multiple ingress gateways that they can name themselves.
One, we want to be able to identify the component type if the name it `foo`.  Two, if they have an ingress gateway named `foo` and a terminating gateway named `foo`, this suffix will prevent name collisions / confusion.

Reply:
Not sure about this. We'll then have everything called `consul-ingress-gateway-ingress-gateway`

Reply:
I don't think most folks use multiple igw's so not sure we should optimize for that. What's the impact if they accidentally set both names to `foo`? Does it fail the helm install/upgrade?

Reply:
By default we have:

```yaml
ingressGateways:
  gateways:
  - name: ingress-gateway
```

so all you need to do in your values is:

```
ingressGateways:
  enabled: true
```

Now this would cause you to have `consul-ingress-gateway-ingress-gateway` everywhere which looks pretty bad imo.

Reply:
Looks like helm install fails before anything...as you inquired about.  Getting errors related to duplicate service accounts.  Getting `serviceaccounts "test-nsy8zi-consul-foo" already exists`.

@lkysow what do you think is the best option? I'm going to look into a helm check at the moment.  I agree with you that this PR is over optimizing for a rare case.  The check is probably going to involve checking:
- look for duplicate terminating gateways 
- look for duplicate ingress gateways
- look for duplicates across all ingress and terminating gateway names.

I'm guessing each of these checks will involve nested loops. So, for us the code will be 🙈 , but I think it will be worth it for a better user UX.  If there are other options or ideas, please let me know what you think.

Reply:
Yeah I think the helm check is a good idea and I apologize in advance for what is likely to be quite un-fun.

Can definitely be a separate PR at least.

Reply:
Cool.  I address it in this PR: https://github.com/hashicorp/consul-k8s/pull/1120

Reply:
That sounds good! Feels like a much better solution than repeated names